Only display 'Symbol' in index when there are terms in its section

tomjw64 opened the pull request:
Stops a hardcoded "Symbol" group container title from overriding the "A" group container title to resolve issue #278.

philschatz commented on rulesets/mixins/_modify.scss:
```diff
@@ -112,7 +112,9 @@
 @mixin modify_addIndexSymbolGroup($location, $name) {
   .os-#{$location}-container > div.group-by:first-of-type > span.group-label {
-    content: $name;
+    &:empty {
```
Woah, that was a weird bug. Is there another way to add this title? Just by looking at the code it does not seem clear to me why `:empty` is added here.

philschatz replied:
Woah, that was a weird bug. Is there another way to add this title? Just by looking at the code it does not seem clear to me why `:empty` is added here.

tomjw64 replied:
When we group index terms, we place `group-by` elements as containers to each group. Each `group-label` contains the first letter by which the terms are sorted. For index terms beginning with not `[a-zA-Z]`, rather than their first letter, we instead group them by an empty string (which in-turn puts it at the top of the index. In the past, it looks like we selected that element by just assuming it was the first label in the index. However, since u-physics contains no terms beginning with a symbol, the `group-by` with the label `A` happened to be the first, and thus was replaced with `Symbol`. Adding this `:empty` will now correctly _only_  select the `group-label` that was meant for the `Symbol` group.

An alternative to using this selector to add the correct title could be instead of grouping symbols by `''`, we could group them by `' '` (or some other `[^a-zA-Z]`, i suppose)  and then select the correct element with a `:match(^[\s])` rather than with an `:empty`. Additionally we could group them by `' Symbols'` (with a space at the beginning to make it appear first, but I think that would make it hard to figure out why it was working.)

Do either of those methods seem preferable, or more clear to a reader?

helenemccarron replied:
Would it make more sense if it was commented in the code to offer an insight on the reasons for selecting this approach?

philschatz replied:
@tomjw64 ooh, I like grouping them by `' Symbols'` (starts with a space) instead. I think that would help make the code clearer

tomjw64 commented:
I made the symbols group sort by `' Symbols'` instead of by `''`, then I changed the label modification selector from `:empty` to a `:match`, to remove the leading space from content. Also updated the documentation comment of the label modification mixin to note the change.
